src/Solr/Query/AbstractQueryBuilder.php

```diff
@@ -70,7 +70,9 @@ public function build()
             $this->getQueryText(),
             0, 
             $this->pagination->getPageSize() * $this->pagination->getCurrentPage(),
-            $this->paramsBuilder->buildAsArray($this->attributetoReset)
+            $this->paramsBuilder
+                ->setAttributeToReset($this->attributetoReset)
```

Comment:
If I understood it correctly, `setAttributesToReset` is only needed in one implementation. I think it should not be implemented in the abstract params builder and probably not even used in the abstract query builder then.

Here we can easily remove the line and move them to `setAttributeToReset()` above:

```diff
     public function setAttributeToReset($attributeToReset)
     {
-        $this->attributetoReset = $attributeToReset;
+        $this->paramsBuilder->setAttributeToReset($attributeToReset);
         return $this;
     }
```

Then move that method down to the class(es) where it is actually needed and we know that the type of `paramsBuilder` supports it: `SearchQueryBuilder` and `CategoryQueryBuilder`(?)